src/GovUk.Education.ExploreEducationStatistics.Data.Services/SubjectMetaService.cs

```diff
@@ -189,37 +197,56 @@ private BoundaryLevel GetBoundaryLevel(GeographicLevel geographicLevel)
 
             if (geoJsonRequested)
             {
-                var codes = observationalUnits.Select(unit => unit is LocalAuthority localAuthority ?
-                    localAuthority.GetCodeOrOldCodeIfEmpty() : unit.Code);
+                var codes = observationalUnits.Select(unit =>
+                    unit is LocalAuthority localAuthority ? localAuthority.GetCodeOrOldCodeIfEmpty() : unit.Code);
 
                 var boundaryLevel = boundaryLevelId ?? GetBoundaryLevel(geographicLevel)?.Id;
 
-                geoJsonByCode = boundaryLevel.HasValue 
+                geoJsonByCode = boundaryLevel.HasValue
                     ? _geoJsonService.Find(boundaryLevel.Value, codes).ToDictionary(g => g.Code)
                     : new Dictionary<string, GeoJson>();
             }
-            
+
             return observationalUnits.Select(observationalUnit =>
             {
-                var value = observationalUnit is LocalAuthority localAuthority ?
-                    localAuthority.GetCodeOrOldCodeIfEmpty() : observationalUnit.Code;
+                var value = observationalUnit is LocalAuthority localAuthority
+                    ? localAuthority.GetCodeOrOldCodeIfEmpty()
+                    : observationalUnit.Code;
 
                 var serializedGeoJson = geoJsonByCode.GetValueOrDefault(value);
-                
-                var geoJson = serializedGeoJson != null 
-                    ? DeserializeGeoJson(serializedGeoJson) 
+
+                var geoJson = serializedGeoJson != null
+                    ? DeserializeGeoJson(serializedGeoJson)
                     : null;
-                
+
                 return new ObservationalUnitGeoJsonMeta
                 {
                     GeoJson = geoJson,
                     Label = observationalUnit.Name,
                     Level = geographicLevel,
-                    Value = value 
+                    Value = value
                 };
             });
         }
 
+        private async Task<Either<ActionResult, Subject>> CheckSubjectExists(Guid subjectId)
```

Comment:
Can you use persistenceHelper.CheckEntityExists<Subject>(subjectId) here instead rather than this full-on method?

Reply:
Done